src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -806,7 +806,7 @@ public void shutdown() {
                     head.setNextOrdered(head);
                     if (! NO_QUEUE_LIMIT) decreaseQueueSize();
                     head = taskNode;
-                    list.add(taskNode.task);
+                    list.add(taskNode.task.handoff());
```

Comment:
When `shutdownNow` is called, one of two caller behaviors is likely:
* They will run the tasks
* They will discard the tasks

I think in both of these cases it's better for the task to have the correct TCCL; I think it's probably OK to also update the counters in the event that the task is actually run (better late than never).  WDYT?

Reply:
> I think in both of these cases it's better for the task to have the correct TCCL

You are correct, the `handoff()` method does retain the tccl by wrapping the delegate task with the general purpose `ContextClassLoaderSavingRunnable`. I'm not sure we want to update counters for this executor when tasks are executed elsewhere, I don't think it should impact the shutdownNow EQE based on whether tasks are discarded or executed.